src/core/user-wallet.ts

```diff
@@ -105,18 +106,28 @@ export abstract class UserWallet {
 
     return rawTransactions.map((item) => {
       const amount = item.credit - item.debit
+      const isCredit = amount > 0
+      const isValidCreditMemo = isCredit && amount < MEMO_SHARING_SATS_THRESHOLD
+      const isDebit = !isCredit
+
       const memoUsername = item.username
-        ? amount > 0
+        ? isCredit
           ? `from ${item.username}`
           : `to ${item.username}`
         : null
 
+      const memoFromPayer = item.memoFromPayer
+        ? isDebit || isValidCreditMemo
```

Comment:
The validation filter is for the receiver, please check.

Reply:
Yup so the intention here is for the filter to happen in `isValidCreditMemo` and this conditional is saying that we _will add_ the memo if it is either a "debit" or a "valid credit"